### PR TITLE
feat: M2-279 - Create Date Picker component

### PR DESCRIPTION
**Pull request by IhorNesterenko99**

### CODE REVIEW

File: src/components/DatePicker/DatePicker.utils.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { format } from 'date-fns';
+
+export const getDatesStringsArray = (value: string) =>
+  value?.split('-').map((element) => element.trim());
+
+export const getStringFromDate = (date: Date) => String(format(date, 'dd MMM yyyy'));
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.utils.ts:6]: maybe it's better to create constant for date format

File: src/components/DatePicker/DatePicker.types.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export type DateVariant = 'start' | 'end';
+
+export type MinMaxDate = 'min' | 'max';
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.types.ts:3]: you can create enums for DateVariant and MinMaxDate and use them

File: src/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import { DateVariant, DatePickerProps, UiType, MinMaxDate } from './DatePicker.types';
+import { DatePickerHeader } from './DatePickerHeader';
+import { getDatesStringsArray, getStringFromDate, getDateFromString } from './DatePicker.utils';
+import { datePlaceholder } from './DatePicker.const';
+
+export const DatePicker = ({ value, setValue, uiType = UiType.oneDate }: DatePickerProps) => {
+  const { t } = useTranslation('app');
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [showFirstCalendar, setShowFirstCalendar] = useState(true);
+  const [showSecondCalendar, setShowSecondCalendar] = useState(false);
```
Comment (kdanilenka32) [inline src/components/DatePicker/DatePicker.tsx:26]: pls rename to visible

File: src/components/DatePicker/DatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    }
+  };
+
+  const getSelectedDate = (variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      return value ? getDateFromString(value) : new Date();
+    } else {
+      let resultVal = new Date();
+      const valueArr = value !== '' && getDatesStringsArray(value);
+      if (valueArr) {
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.tsx:70]: if (valueArr) {
 const index = variant === 'start' ? 0 : 1;
resultVal = valueArr[index] !== datePlaceholder ? getDateFromString(valueArr[index]) : new Date().... }

as a variant)

File: src/components/DatePicker/DatePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        setValue(`${startDate || datePlaceholder} - ${endDate || datePlaceholder}`);
+      }
+    }
+  };
+
+  const getSelectedDate = (variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      return value ? getDateFromString(value) : new Date();
+    } else {
+      let resultVal = new Date();
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.tsx:68]: maybe it's better to rename resultVal to "selectedDate" and valueArr to "dateArray"

File: src/components/DatePicker/DatePicker.utils.ts (hunk 1 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.utils.ts:6]: done

File: src/components/DatePicker/DatePicker.tsx (hunk 3 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:26]: done

File: src/components/DatePicker/DatePicker.tsx (hunk 4 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:70]: done

File: src/components/DatePicker/DatePicker.tsx (hunk 5 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:68]: done

File: src/components/DatePicker/DatePicker.types.ts (hunk 2 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.types.ts:3]: done

File: src/components/DatePicker/DatePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    setSecondCalendarVisible((prevState) => !prevState);
+  };
+
+  const handleDateChange = (date: Date | null, variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      setValue(date ? getStringFromDate(date) : '');
+    } else {
+      const datesArr = getDatesStringsArray(value);
+      if (date) {
+        const startDate = variant === 'start' ? getStringFromDate(date) : datesArr?.[0];
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.tsx:56]: use DateVariant enum

File: src/components/DatePicker/DatePicker.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+  const getDayPickerCalendar = (variant?: DateVariant) => (
+    <ReactDatePicker
+      selected={getSelectedDate(variant)}
+      onChange={(date) => handleDateChange(date, variant)}
+      inline
+      formatWeekDay={(nameOfDay) => nameOfDay.slice(0, 1)}
+      renderCustomHeader={(props) => <DatePickerHeader uiType={uiType} {...props} />}
+      minDate={variant === 'end' ? getMinMaxDate(MinMaxDate.min) : undefined}
+      maxDate={variant === 'start' ? getMinMaxDate(MinMaxDate.max) : undefined}
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.tsx:97]: DateVariant enum

File: src/components/DatePicker/DatePicker.tsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+  };
+
+  const getSelectedDate = (variant?: DateVariant) => {
+    if (uiType === UiType.oneDate) {
+      return value ? getDateFromString(value) : new Date();
+    } else {
+      let selectedDate = new Date();
+      const datesArr = value !== '' && getDatesStringsArray(value);
+      if (datesArr) {
+        const index = variant === 'start' ? 0 : 1;
```
Comment (wildkiomi) [inline src/components/DatePicker/DatePicker.tsx:71]: DateVariant enum

File: src/components/DatePicker/DatePicker.tsx (hunk 6 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:56]: done

File: src/components/DatePicker/DatePicker.tsx (hunk 7 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:97]: done

File: src/components/DatePicker/DatePicker.tsx (hunk 8 above)
Comment (IhorNesterenko99) [reply src/components/DatePicker/DatePicker.tsx:71]: done

**Review by wildkiomi — APPROVED** [review]

**Pull request merged**